### PR TITLE
Unblock supported compression libs in pytables

### LINKED ISSUE #14478: df.to_hdf() blocks some supported pytables compression types ‘blosc:lz4’, ‘blosc:lz4hc’, ‘blosc:snappy’, ‘blosc:zlib’ and ‘blosc:zstd’
`df.to_hdf()` blocks access to the following compressors offered in pytables 3.3.0:  ‘blosc:lz4’, ‘blosc:lz4hc’, ‘blosc:snappy’, ‘blosc:zlib’ and ‘blosc:zstd’. 

I would like to try blosc:lz4 compression for some of the bigger data I have to compare size and speed to LZO.

``` python
df.to_hdf(path, 'df', complib='blosc:lz4')
D:\Python27\lib\site-packages\pandas\io\pytables.pyc in __init__(self, path, mode, complevel, complib, fletcher32, **kwargs)
    434 
    435         if complib not in (None, 'blosc', 'bzip2', 'lzo', 'zlib'):
--> 436             raise ValueError("complib only supports 'blosc', 'bzip2', lzo' "
    437                              "or 'zlib' compression.")
    438 

ValueError: complib only supports 'blosc', 'bzip2', lzo' or 'zlib' compression.

```

### PR DESCRIPTION
**Pull request by linebp**
- [x] closes #14478
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/test_pytables.py [hunk 1]
```diff
@@ -736,6 +736,114 @@ def test_put_compression_blosc(self):
             store.put('c', df, format='table', complib='blosc')
             tm.assert_frame_equal(store['c'], df)
 
+    def test_complibs(self):
+        # Test data
+        data = [{'ID': '7082-GTZ-572490',
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: don't specify data like this, just use ``tm.makeDataFrame``

File: pandas/tests/io/test_pytables.py [hunk 2]
```diff
@@ -736,6 +736,114 @@ def test_put_compression_blosc(self):
...
+                 'timestamp': datetime.datetime(2017, 7, 14, 14, 11, 39)}]
+        df = pd.DataFrame.from_dict(data)
+
+        # Building list of all complibs and complevels tuples
+        all_complibs = tables.filters.all_complibs
+        all_levels = range(0, 10)
+        all_tests = [(lib, lvl) for lib in all_complibs for lvl in all_levels]
+
+        for (lib, lvl) in all_tests:
+            with ensure_clean_store(self.path, complib=lib,
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: I would rather you use ``.to_hdf`` and ``.read_hdf`` here

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
this is going to need a release note for 0.20.1 (but doesn't exist as of yet, soon).

File: pandas/tests/io/test_pytables.py [hunk 3]
```diff
@@ -736,6 +736,20 @@ def test_put_compression_blosc(self):
             store.put('c', df, format='table', complib='blosc')
             tm.assert_frame_equal(store['c'], df)
 
+    def test_complibs(self):
+        df = tm.makeDataFrame()
+
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: can you add a test that validates that ``complevel=0`` turns off compression?

File: pandas/tests/io/test_pytables.py [hunk 4]
```diff
@@ -736,6 +736,20 @@ def test_put_compression_blosc(self):
             store.put('c', df, format='table', complib='blosc')
             tm.assert_frame_equal(store['c'], df)
 
+    def test_complibs(self):
+        df = tm.makeDataFrame()
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: add a comment referencing the issue

File: pandas/tests/io/test_pytables.py [hunk 5]
```diff
@@ -4942,7 +4956,7 @@ def test_invalid_complib(self):
                        columns=list('ABCDE'))
         with ensure_clean_path(self.path) as path:
             pytest.raises(ValueError, df.to_hdf, path,
```
Comment (jreback) [inline pandas/tests/io/test_pytables.py]: this needs to have be

```
with pytest.raises(ValueError):
   ...
```

File: pandas/tests/io/test_pytables.py (hunk 3 above)
Comment (linebp) [inline pandas/tests/io/test_pytables.py]: How do I best do that? There is the h5dump utility that can tell me if its compressed or I can compare the size of the files perhaps?

Comment (jreback) [inline pandas/tests/io/test_pytables.py]: look in the meta data of the file
lots of _ attributes that define things like this
pytables docs might be helpful

**Comment by linebp** [comment]
It turned out that not all configurations has lzo compression available, so I removed those tests for those scenarios. 

Is this looking good? The whatsnew file for 20.1 is still not in existence?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@linebp put this in 0.20.2 Enhancements (0.20.1 doesn't exist and was a combine release with 0.20.0).

ping on green.

File: pandas/core/generic.py [hunk 6]
```diff
@@ -1267,12 +1267,17 @@ def to_hdf(self, path_or_buf, key, **kwargs):
...
-        complib : {'zlib', 'bzip2', 'lzo', 'blosc', None}, default None
-            If complevel is > 0 apply compression to objects written
-            in the store wherever possible
+        complevel : int, 0-9, default 0
+            Specifies a compression level for data.
+            A value of 0 disables compression.
+        complib : {'zlib', 'lzo', 'bzip2', 'blosc', None}, default None
+            Specifies the compression library to be used.
+            Additional compressors for Blosc are supported (default if no
+            compressor specified: 'blosc:blosclz'):
```
Comment (jreback) [inline pandas/core/generic.py]: add as of 0.20.2 these addtional compressors are available.

File: pandas/io/pytables.py [hunk 7]
```diff
@@ -402,12 +402,17 @@ class HDFStore(StringMixin):
...
-    complib : {'zlib', 'bzip2', 'lzo', 'blosc', None}, default None
-            If complevel is > 0 apply compression to objects written
-            in the store wherever possible
+    complevel : int, 0-9, default 0
+            Specifies a compression level for data.
+            A value of 0 disables compression.
+    complib : {'zlib', 'lzo', 'bzip2', 'blosc', None}, default None
+            Specifies the compression library to be used.
+            Additional compressors for Blosc are supported (default if no
+            compressor specified: 'blosc:blosclz'):
```
Comment (jreback) [inline pandas/io/pytables.py]: same here

**Comment by linebp** [comment]
Its all green now.

**Comment by jreback** [comment]
thanks!

**Pull request merged**